### DISCUSSION TITLE
mboot versioning

### DISCUSSION THREAD

**Discussion by victorallume** [micropython/micropython#15768] [STM32 / Pyboard]
Is there any way to determine what version of mboot is on a device from within micropython? The `fwupdate.py` file allows me to check if mboot is the same a supplied dfu file, but I can't find anything about versioning (as far as I can tell there is no versioning of mboot other than the micropython version it comes from). 

I ask this as with the update of LFS2 in micropython v1.22.0, mboot needs to be updated first to a version that can handle the newer fs format. So as part of our OTA process, we want to check if mboot is greater than a certain version. 

If this is not currently possible, what would be involved in doing this? Store the version in a known flash location, and have micropython read this?

**Comment by victorallume** [comment]
I've created a PR at https://github.com/micropython/micropython/pull/15874

**Reply by Josverl** [reply]
Would it be clearer to use the mpy version + git hash, that is also used for the micropython build version itself?

That would be simpler to compare for mere mortals.
There is already code for that in the tools folder, (but can't find it from my phone)

**Comment by bidrohini** [comment]
You can access flash memory directly using the machine module. For example, you can use the machine.mem32 or machine.mem8 to read specific memory addresses. You can add a function to read from the address where the version is stored.

**Reply by victorallume** [reply]
Yes, but the problem was that the version is not stored anywhere, and making sure it is at a well-known memory location is tricky. My PR addresses this.
